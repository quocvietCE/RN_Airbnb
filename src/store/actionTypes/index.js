export const AUTH_ACTION = {
  USER_LOGIN: 'USER_LOGIN',
  USER_LOGIN_SUCCESS: 'USER_LOGIN_SUCCESS',
  USER_LOGIN_FAILURE: 'USER_LOGIN_FAILURE',

  LOG_OUT: 'LOG_OUT',
};

export const APP_ACTION = {
  UPDATE_LANGUAGE: 'UPDATE_LANGUAGE',
  REQUIRE_LOGIN_PAGE: 'REQUIRE_LOGIN_PAGE',
  UPDATE_POSITION: 'UPDATE_POSITION',
  OPEN_MODAL: 'OPEN_MODAL',
  CLOSE_MODAL: 'CLOSE_MODAL',
  OPEN_ADDRESS_MODAL: 'OPEN_ADDRESS_MODAL',
  CLOSE_ADDRESS_MODAL: 'CLOSE_ADDRESS_MODAL',
  CHANGED_COUNTRY: 'CHANGED_COUNTRY',
  CHANGE_COUNTRY_SUCCESS: 'CHANGE_COUNTRY_SUCCESS',
  SET_APP_DATA: 'SET_APP_DATA',
  GET_CONFIGURATION_APP: 'GET_CONFIGURATION_APP',
  GET_CONFIGURATION_APP_SUCCESS: 'GET_CONFIGURATION_APP_SUCCESS',
  GET_CONFIGURATION_APP_FAIL: 'GET_CONFIGURATION_APP_FAIL',
  CLEAR_ERROR: 'CLEAR_ERROR',
  OPEN_FULL_MAP_VIEW: 'OPEN_FULL_MAP_VIEW',
  CLOSE_FULL_MAP_VIEW: 'CLOSE_FULL_MAP_VIEW',
  SAVE_PERMISSION_PHOTO: 'SAVE_PERMISSION_PHOTO',
};

export const LISTING_ACTION = {
  GET_LISTING_DATA: 'GET_LISTING_DATA',
  GET_LISTING_DATA_SUCCESS: 'GET_LISTING_DATA_SUCCESS',
  GET_LISTING_DATA_FAILURE: 'GET_LISTING_DATA_FAILURE',
};

export const NETWORK_ACTION = {
  SET_STATUS_CONNECTED: 'SET_STATUS_CONNECTED',
  SET_STATUS_CONNECTED_SUCCESS: 'SET_STATUS_CONNECTED_SUCCESS',
  SET_STATUS_CONNECTED_FAILURE: 'SET_STATUS_CONNECTED_FAILURE',
};
