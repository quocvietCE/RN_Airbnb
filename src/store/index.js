import {createStore, applyMiddleware, compose} from 'redux';
import createSagaMiddleware from 'redux-saga';
import {persistStore, persistReducer} from 'redux-persist';
import AsyncStorage from '@react-native-community/async-storage';
import logger from 'redux-logger';
import rootSaga from './sagas';

import rootReducer from './reducers';

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
  // blacklist: ['auth', 'app'],
  blacklist: ['app, listing'],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);
const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const sagaMiddleware = createSagaMiddleware();

export const store = createStore(
  persistedReducer,
  composeEnhancer(applyMiddleware(sagaMiddleware, logger)),
);

export const persistor = persistStore(store);

sagaMiddleware.run(rootSaga);
export default () => ({store, persistor});
