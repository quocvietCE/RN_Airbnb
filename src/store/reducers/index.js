import {combineReducers} from 'redux';
import auth from './authReducer';
import app from './appReducer';
import listing from './listingReducer';
import network from './networkReducer';

export default combineReducers({
  auth,
  app,
  listing,
  network,
});
