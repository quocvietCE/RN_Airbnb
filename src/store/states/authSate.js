const initState = {
  resLogin: {},
  account: null,
  token: null,

  loggedInState: null,
};

const authState = {
  initState,
};

export default authState;
