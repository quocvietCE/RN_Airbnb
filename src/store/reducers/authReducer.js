import {AUTH_ACTION} from '../actionTypes';
import authState from '../states/authSate';

const authReducer = (state = authState.initState, action) => {
  switch (action.type) {
    case AUTH_ACTION.USER_LOGIN_SUCCESS:
      return {
        ...state,
        loggedInState: action.result,
      };
    case AUTH_ACTION.LOG_OUT:
      return {
        ...authState.initState,
      };
    default:
      return state;
  }
};

export default authReducer;
