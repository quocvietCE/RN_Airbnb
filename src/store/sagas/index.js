import {takeLatest, all, takeEvery} from 'redux-saga/effects';
import authSaga from './authSaga';
import {AUTH_ACTION} from '../actionTypes';

function* handleLogin() {
  yield takeLatest(AUTH_ACTION.USER_LOGIN, authSaga.handleLogin);
}

export default function* rootSaga() {
  yield all([handleLogin()]);
}
