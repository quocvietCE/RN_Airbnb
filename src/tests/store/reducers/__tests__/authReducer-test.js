import React from 'react';
import 'react-native';
import authReducer from '../../../../store/reducers/authReducer';
import authSate from '../../../../store/states/authSate';
import {AUTH_ACTION} from '../../../../store/actionTypes';

describe('Test appReducer', () => {
  it('returns the same appReducer state on an unhandled state', () => {
    expect(authReducer(authSate.initState, {type: '_NULL'})).toMatchSnapshot();
  });

  it('Check Snapshot AUTH_ACTION.USER_LOGIN_SUCCESS type', () => {
    expect(
      authReducer(authSate.initState, {
        type: AUTH_ACTION.USER_LOGIN_SUCCESS,
      }),
    ).toMatchSnapshot();
  });

  it('Check Snapshot AUTH_ACTION.USER_LOGIN_FAILURE type', () => {
    expect(
      authReducer(authSate.initState, {
        type: AUTH_ACTION.USER_LOGIN_FAILURE,
        error: 'Network Error',
      }),
    ).toMatchSnapshot();
  });

  it('handles login AUTH_ACTION.USER_LOGIN_SUCCESS type', () => {
    expect(
      authReducer(authSate.initState, {
        type: AUTH_ACTION.USER_LOGIN_SUCCESS,
        result: true,
      }),
    ).toEqual({
      ...authSate.initState,
      loggedInState: true,
    });
  });

  it('handles logout AUTH_ACTION.LOG_OUT type', () => {
    expect(
      authReducer(authSate.initState, {
        type: AUTH_ACTION.LOG_OUT,
      }),
    ).toEqual({
      ...authSate.initState,
    });
  });
});
