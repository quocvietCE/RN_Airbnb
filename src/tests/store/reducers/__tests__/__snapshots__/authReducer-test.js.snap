// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test appReducer Check Snapshot AUTH_ACTION.USER_LOGIN_FAILURE type 1`] = `
Object {
  "loggedInState": null,
}
`;

exports[`Test appReducer Check Snapshot AUTH_ACTION.USER_LOGIN_SUCCESS type 1`] = `
Object {
  "loggedInState": undefined,
}
`;

exports[`Test appReducer returns the same appReducer state on an unhandled state 1`] = `
Object {
  "loggedInState": null,
}
`;
